packages/consumption/src/modules/attributes/AttributesController.ts

```diff
@@ -322,6 +322,11 @@ export class AttributesController extends ConsumptionBaseController {
         this.eventBus.publish(new AttributeDeletedEvent(this.identity.address.toString(), attribute));
     }
 
+    public async deleteAttributesSharedWithPeer(peer: ICoreAddress): Promise<void> {
```

Comment:
`peer` should be a `CoreAddress`, otherwise `.toString()` won't work unless peer is actually of type `CoreAddress`.

```ts
{ address: "id1234" }.toString()
// [object Object]
```

If you want to keep the interface type for `peer` use `peer.address` to get the actual address as a string.